app/frontend/actions/delegate.ts

```diff
@@ -64,59 +70,64 @@ export default (store: Store) => {
     const poolHash = state.shelleyDelegation?.selectedPool?.poolHash as string
     const isStakingKeyRegistered = getSourceAccountInfo(state).shelleyAccountInfo.hasStakingKey
     const stakingAddress = getSourceAccountInfo(state).stakingAddress
-    const txPlanResult = await prepareTxPlan({
-      poolHash,
-      stakingAddress,
-      isStakingKeyRegistered,
-      txType: TxType.DELEGATE,
-    })
-    const newState = getState()
-    if (hasPoolIdentifiersChanged(newState)) {
-      return
-    }
-    const balance = getSourceAccountInfo(state).balance as Lovelace
-
-    if (txPlanResult.success === true) {
-      setState({
-        shelleyDelegation: {
-          ...newState.shelleyDelegation,
-          delegationFee: (txPlanResult.txPlan.fee + txPlanResult.txPlan.deposit) as Lovelace,
-        },
+    if (stakingAddress && poolHash) {
```

Comment:
I'd rather put here a runtime assertion that stakingAddress must not be null as the original logic apparently relied on that and I don't see it working properly without that assumption being fulfilled and it doesn't really make sense to delegate without the staking address being set

Reply:
Since this logic will be repeated in three places does it make sense to throw this error in `prepareTxPlan` and relax the `stakingAddress` type in the signature?

Reply:
I looked deeper into that and the only place where we set the stakingKey to null is the initial state of the app where we set an initial value to the `accountsInfo` property, but that doesn't really make sense. Let's just set `accountsInfo` there to an empty array and then we should be good to enforce that stakingAddress must be always defined

I mean here: https://github.com/vacuumlabs/adalite/blob/3119857be681ba54376a032b215ffa9544e5474b/app/frontend/state.ts#L223

Do you see some reason @PeterBenc why we would have to set some default "null" account in the initial state of the app, or is that just legacy from the times when we had a single account in the state?

Reply:
Agreed, I dont see any reason why `accountInfos` cant be initially `[]`. Its not used until the wallet is loaded and then just assigned. 

BUT, I think I remember that setting account info to `[]` was actually a problem when logging out. I think that after logging out, some components still (for a brief moment I guess) "used" `accountInfos[state.activeAccount]` and accessed its keys so errors like `cant... something of undefined` popped up and the page crashed. Surely that probably points to an issue on its own.

So that was probably the reason for keeping it the way it is now, but I just tried setting it to `[]` and it works as expected. So I guess I either remember it wrong the issue causing it got fixed.

Anyway, just something to consider but a bit unrelated to this, having the initial `accountsInfo` as `[]` seem to work and makes sense.